src/uu/lslocks/src/lslocks.rs

```diff
@@ -0,0 +1,306 @@
+// This file is part of the uutils util-linux package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::{fmt, fs, str::FromStr};
+
+use clap::{crate_version, Command};
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+// See https://www.man7.org/linux/man-pages/man5/proc_locks.5.html for details on each field's meaning
+#[derive(Debug)]
+struct Lock {
+    _ord: usize,
+    lock_type: LockType,
+    mandatory: bool,
```

Comment:
For consistency reasons I would handle it like the `mode` property. Something like:
```suggestion
    status: LockStatus,
```

Reply:
Yep, my initial implementation had an enum for this aswell but then I noticed that even the human-readable representation for this is just a `1` or `0` so I switched to a simple binary value instead.